### PR TITLE
Merge MultiApiRobolectricTestRunner into RobolectricTestRunner.

### PR DESCRIPTION
**Pull request by xian**
### Overview
Merge MultiApiRobolectricTestRunner functionality into RobolectricTestRunner.

### Proposed Changes
RobolectricTestRunner will run tests on multiple SDKs if requested.
Moved Config calculation and SDK selection to SdkPicker (need better name).

### To Do
- [x] Make `@Config(sdk=…)` and `@Config(minSdk=…,maxSdk=…)` mutually exclusive at any given level of configuration (e.g. per test method, class, or properties file), and mutually overriding between levels.
- [x] By default, tests should still run against `targetSdkLevel` from the manifest.
- [x] Support `@Config(sdk=ALL_SDKS)` and `@Config(sdk=LATEST_SDK)`.
- [x] If `targetSdkVersion` is unspecified, use `minSdkVersion`.
- [x] If `minSdkVersion` and `maxSdkVersion` are specified, clip to supported SDKs.
- [ ] Maybe find a better name for `SdkPicker`?
- [ ] Docs and javadoc

### CODE REVIEW

File: robolectric-annotations/src/main/java/org/robolectric/annotation/DefaultApplication.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.robolectric.annotation;
+
+import android.app.Application;
+
+class DefaultApplication extends Application {
```
Comment (jongerrish) [inline robolectric-annotations/src/main/java/org/robolectric/annotation/DefaultApplication.java]: Cool, I realized this was what I should have done in the first place

File: robolectric/src/main/java/org/robolectric/ConfigMerger.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      Properties properties = new Properties();
+      properties.load(resourceAsStream);
+      return properties;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @NotNull
+  private List<String> packagesFor(Class<?> javaClass) {
```
Comment (jongerrish) [inline robolectric/src/main/java/org/robolectric/ConfigMerger.java]: packageHeirarchyOf() might be a better name?

File: robolectric/src/test/java/org/robolectric/ConfigMergerTest.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+      }
+    }.getConfig(testClass, info, Config.Builder.defaults().build());
+  }
+
+  private Config configFor(Class<?> testClass, String methodName) throws InitializationError {
+    Method info = getMethod(testClass, methodName);
+    return new ConfigMerger().getConfig(testClass, info, Config.Builder.defaults().build());
+  }
+
+  private Method getMethod(Class<?> testClass, String methodName) {
```
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/ConfigMergerTest.java]: Can be static

File: robolectric/src/test/java/org/robolectric/TestRunners.java [hunk 4]
```diff
@@ -2,56 +2,37 @@
...
+      return new Config.Builder(super.buildGlobalConfig())
+          .setManifest(resourceFile("TestAndroidManifest.xml").toString())
+          .build();
     }
   }
 
-  public static class MultiApiWithDefaults extends MultiApiRobolectricTestRunner {
-
-    public MultiApiWithDefaults(Class<?> testClass) throws Throwable {
+  public static class MultiApiSelfTest extends SelfTest {
```
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/TestRunners.java]: Could we just make this SelfTest then any of the existing usages of SelfTest would get the Multi Api behaviour? Any that don't work on all APIs e.g: an existing bug, we could control with an @Config(sdk = ) ?

Comment (xian) [inline robolectric/src/test/java/org/robolectric/TestRunners.java]: Sure, I'll open an issue for that so we can address it separately.

Comment (xian) [inline robolectric/src/test/java/org/robolectric/TestRunners.java]: #2775

**Comment by xian** [comment]
Docs underway at http://robolectric.org/configuring/

**Pull request merged**